building/qubes-r3-building.md

```diff
@@ -24,36 +38,80 @@ $ wget https://keys.qubes-os.org/keys/qubes-master-signing-key.asc
 $ gpg --import qubes-master-signing-key.asc 
 $ gpg --edit-key 36879494
 # Verify fingerprint!, set trust to *ultimate*
+# (Typical sequence is fpr, trust, 5, q)
 $ wget https://keys.qubes-os.org/keys/qubes-developers-keys.asc
 $ gpg --import qubes-developers-keys.asc
 ~~~
 
-Note we do *not* relay above on the security of our server (keys.qubes-os.org) nor the connection (ssl, cert) -- we only rely on you getting the Qubes Master Signing Key fingerprint *somehow* and ensure they match!
+And if you will be building the Whonix templates:
+
+~~~
+$ wget https://github.com/QubesOS/qubes-builder-debian/blob/master/keys/whonix-developer-patrick.asc
+$ gpg --import whonix-developer-patrick.asc
```

Comment:
I'd defer this step to later point, where builder-debian is already downloaded and its signature is verified. Otherwise you rely here on just https + github.

Reply:
How is this different in concept than the 
```
$ wget https://keys.qubes-os.org/keys/qubes-master-signing-key.asc
$ gpg --import qubes-master-signing-key.asc 

$ wget https://keys.qubes-os.org/keys/qubes-developers-keys.asc
$ gpg --import qubes-developers-keys.asc
```
steps?
I'm trying to do all the keyring stuff at once.

Reply:
Here you're importing key without verification if it is really the right one. When you have it downloaded by qubes-builder, it has verified that the file is covered by properly signed git tag.
Note that for the master key we have specific step of fingerprint verification. If we add such step also for the other key, it would be ok.

Reply:
Also, keys in qubes-developers-keys.asc are signed by master key (and we use that information later), but whonix key is not.

Reply:
I understand both your points, thank you. Maybe the better way to handle this would be to make the script more robust instead of (or in addition to) in the documentation. I'll take a look and update this when I have a better idea.